token/program2/src/instruction.rs

```diff
@@ -121,34 +122,37 @@ pub enum TokenInstruction {
     ///   1. '[]' The source account's multisignature owner.
     ///   2. ..2+M '[signer]' M signer accounts
     Revoke,
-    /// Sets a new owner of a mint or account.
+    /// Sets a new authority of a mint or account.
     ///
     /// Accounts expected by this instruction:
     ///
-    ///   * Single owner
-    ///   0. `[writable]` The mint or account to change the owner of.
-    ///   1. `[]` The new owner/delegate/multisignature.
-    ///   2. `[signer]` The owner of the mint or account.
+    ///   * Single authority
+    ///   0. `[writable]` The mint or account to change the authority of.
+    ///   1. `[]` The new authority/multisignature.
+    ///   2. `[signer]` The current authority of the mint or account.
     ///
-    ///   * Multisignature owner
-    ///   0. `[writable]` The mint or account to change the owner of.
-    ///   1. `[]` The new owner/delegate/multisignature.
-    ///   2. `[]` The mint's or account's multisignature owner.
+    ///   * Multisignature authority
+    ///   0. `[writable]` The mint or account to change the authority of.
+    ///   1. `[]` The new authority/multisignature.
+    ///   2. `[]` The mint's or account's multisignature authority.
     ///   3. ..3+M '[signer]' M signer accounts
-    SetOwner,
+    SetAuthority {
+        /// The type of authority to update.
+        authority_type: AuthorityType,
```

Comment:
I don't like the inconsistently between `InitializeMint` and `SetAuthority`.  `InitializeMint` accepts a `COption<Pubkey>` in the instruction data for the authority, here it's passed as an account key.

I think `SetAuthority` should accept a `COption<Pubkey>`.  This allows us to very clearly remove the mint/freeze authority by providing `None`